src/engine/TransitivePathImpl.h

```diff
@@ -153,35 +163,30 @@ class TransitivePathImpl : public TransitivePathBase {
     auto [startSide, targetSide] = decideDirection();
     std::shared_ptr<const Result> subRes = subtree_->getResult();
```

Comment:
Can you add an explicit `false` here, and add a comment that we always need the full materialized result?